cmd/pyroscope/command/adhoc.go

```diff
@@ -0,0 +1,55 @@
+package command
+
+import (
+	"github.com/spf13/cobra"
+
+	"github.com/pyroscope-io/pyroscope/pkg/adhoc"
+	"github.com/pyroscope-io/pyroscope/pkg/cli"
+	"github.com/pyroscope-io/pyroscope/pkg/config"
+)
+
+func newAdhocCmd(cfg *config.Adhoc) *cobra.Command {
+	vpr := newViper()
+
+	cmd := &cobra.Command{
+		Use:   "adhoc [flags] [args]",
+		Short: "Profile a process and save the results to be used in adhoc mode",
+		Long: `adhoc command is a complete toolset to profile a process and save the profiling
```

Comment:
A proposal of how to integrate the different profiling modes with the `adhoc` command in the CLI.